test/powershell/Modules/Microsoft.PowerShell.Utility/Sort-Object.Tests.ps1

```diff
@@ -222,3 +222,229 @@ Describe "Sort-Object DRT Unit Tests" -Tags "CI" {
 		$results[2] | Should Be 1
 	}
 }
+
+Describe 'Sort-Object Top and Bottom Unit Tests' -Tags 'CI' {
+
+	# Helper function to compare two sort entries
+	function Compare-SortEntry
+	{
+		param($nSortEntry, $fullSortEntry)
+		if ($nSortEntry -is [System.Array]) {
+			# Arrays must be compared differently than everything else
+			$nSortEntry.Equals($fullSortEntry) | Should Be $true
+		} else {
+			$nSortEntry | Should Be $fullSortEntry
+		}
+	}
+
+	# Helper function that compares a full sort with an n-sort
+	function Test-SortObject {
+		param([array]$unsortedData, [hashtable]$baseSortParameters, [string]$nSortType, [int]$nValue)
+		$nSortParameters = @{
+			$nSortType = $nValue
+		}
+		# Sort the data
+		$fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+		$nSortResults = $unsortedData | Sort-Object @baseSortParameters @nSortParameters
+		# Verify the counts when not doing a -Unique sort
+		if (-not $baseSortParameters.ContainsKey('Unique')) {
+			$nSortResults.Count | Should Be $(if ($nSortParameters[$nSortType] -gt $unsortedData.Length) {$unsortedData.Length} else {$nSortParameters[$nSortType]})
+			$fullSortResults.Count | Should Be $unsortedData.Length
+		}
+		# Compare the n-sort result entries with their corresponding full sort result entries
+		if ($nSortType -eq 'Top') {
+			$range = 0..$($nSortResults.Count - 1)
+		} else {
+			$range = -$nSortResults.Count..-1
+		}
+		foreach ($i in $range) {
+			Compare-SortEntry $nSortResults[$i] $fullSortResults[$i]
+		}
+	}
+
+	# Test cases when only the n-sort type needs to be changed
+	$topBottom = @(
+		@{nSortType='Top'   }
+		@{nSortType='Bottom'}
+	)
+
+	# Test cases when the n-sort type and the order type need to be changed
+	$topBottomAscendingDescending = @(
+		@{nSortType='Top';    orderType='ascending' }
+		@{nSortType='Top';    orderType='descending'}
+		@{nSortType='Bottom'; orderType='ascending' }
+		@{nSortType='Bottom'; orderType='descending'}
+	)
+
+	Context 'Integer n-sort' {
+
+		$unsortedData = 973474993,271612178,-1258909473,659770354,1829227828,-1709391247,-10835210,-1477737798,1125017828,813732193
+
+		It 'Return the <nSortType> N sorted in <orderType> order' -TestCases $topBottomAscendingDescending {
+			param([string]$nSortType, [string]$orderType)
+			$baseSortParameters = if ($orderType -eq 'Descending') {@{Descending = $true}} else {@{}}
```

Comment:
As long as you're splatting, you can simplify with something like:

```powershell
... = @{Descending = $orderType -eq 'descending'}
```

And you decide to change this, you might as well fix the case to be consistent.